backend/src/zac/api/permissions.py

```diff
@@ -65,3 +67,38 @@ def has_permission(self, request: Request, view: APIView) -> bool:
 
         zaak_url = serializer.validated_data[self.zaak_attr]
         return self._has_zaak_permission(request, zaak_url)
+
+
+class DefinitionBasePermission(permissions.BasePermission):
+    permission: Permission
+    object_type: str = PermissionObjectType.zaak
+
+    def __new__(cls, *args, **kwargs):
+        permission = getattr(cls, "permission", None)
+        if permission is None:
+            raise ImproperlyConfigured(
+                "%s is missing the 'permission' attribute" % cls.__name__
+            )
+        return super().__new__(cls, *args, **kwargs)
+
+    def has_object_permission(self, request, view, obj):
+        permission_definitions = (
+            PermissionDefinition.objects.for_user(request.user)
+            .actual()
+            .filter(permission=self.permission)
+        )
+
+        # first check atomic permissions
+        if permission_definitions.filter(object_url=obj.url).exists():
+            return True
+
+        # then check blueprint permissions
+        for permission in (
+            permission_definitions.filter(object_type=self.object_type)
+            .exclude(policy={})
```

Comment:
Since we're not dealing with atomic permissions here anymore, and `policy` can be "empty" but also `null` (see your models and review of other PR), this looks a bit odd. How are values handled that are `null`? And - if there's no `object_url`, that makes it a policy definition because of the mutual exclusivity, so shouldn't that mean that a policy MUST be entered?

```suggestion
            .filter(object_url="")
```

Reply:
you are right, thanks for explanation